.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+See the [Scientific Python Developer Guide][spc-dev-intro] for a detailed
+description of best practices for developing scientific packages.
+
+[spc-dev-intro]: https://learn.scientific-python.org/development/
+
+# Commit and pull request expectations
+
+- Every commit should reference the issue number with which it is associated.
+- Commits should be reasonably granular and semantically atomic.
+- Pull requests should not be squashed upon merging.
```

Comment:
I added these guidelines to the `CONTRIBUTING.md` to help ensure requirements traceability for this project. Traceability will make it much easier to produce documentation for the FDA in the future. Please see if you like this approach. We could also enforce this programmatically by making a pre-commit hook that ensures every commit references some issue number. Let me know if this sounds good. If so then we can add the same guidelines and pre-commit hook to the Slicer software repos.